dotcom-rendering/src/lib/tuple.test.ts

```diff
@@ -0,0 +1,83 @@
+import { takeFirst } from './tuple';
+
+describe('takeFirst', () => {
+	it('Always returns the correct array length when the array is one less, the same as, or one more than n', () => {
```

Comment:
This is really asking for [`it.each`](https://jestjs.io/docs/api#describeeachtablename-fn-timeout)

```ts
it.each([
 // take, expected length, array
 [1, 1, [0, 1]],
 [1, 1, [0]],
 [1, 0, []],
 // etc.
])("Can take %s correctly", (count, expectedLength, array) => {
		expect(takeFirst(array, count).length).toEqual(expectedLength);
}
```